Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPISidebar.mm

```diff
@@ -0,0 +1,243 @@
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+#import "WebExtensionUtilities.h"
+
+#if ENABLE(WK_WEB_EXTENSIONS_SIDEBAR)
+
+#import <WebKit/WKPreferencesPrivate.h>
+#import <WebKit/_WKFeature.h>
+
+
+namespace TestWebKitAPI {
+
+static constexpr auto *sidebarActionManifest = @{
+    @"manifest_version": @3,
+    @"name": @"SidebarAction Test",
+    @"description": @"SidebarAction Test",
+    @"version": @"1",
+
+    @"permissions": @[],
+    @"background": @{
+        @"scripts": @[ @"background.js", ],
+        @"type": @"module",
+        @"persistent": @NO,
+    },
+
+    @"sidebar_action": @{
+        @"default_title": @"Test Sidebar",
+        @"default_panel": @"sidebar.html",
+        @"default_icon": @{
+            @"16": @"sidebar-16.png",
+            @"32": @"sidebar-32.png",
+        },
+        @"open_at_install": @NO,
+    },
+};
+
+
+static auto *sidePanelManifest = @{
+    @"manifest_version": @3,
+
+    @"name": @"SidePanel Test",
+    @"description": @"SidePanel Test",
+    @"version": @"1",
+
+    @"permissions": @[ @"sidePanel" ],
+    @"background": @{
+        @"scripts": @[ @"background.js" ],
+        @"type": @"module",
+        @"persistent": @NO,
+    },
+};
+
+static auto *sidebarActionAndPanelManifest = @{
+    @"manifest_version": @3,
+
+    @"name": @"Sidebar Test (both)",
+    @"description": @"Sidebar Test (both)",
+    @"version": @"1",
+
+    @"permissions": @[ @"sidePanel" ],
+
+    @"background": @{
+        @"scripts": @[ @"background.js" ],
+        @"type": @"module",
+        @"persistent": @NO,
+    },
+
+    @"sidebar_action": @{
+        @"default_title": @"Test Sidebar",
+        @"default_panel": @"sidebar.html",
+        @"default_icon": @{
+            @"16": @"sidebar-16.png",
+            @"32": @"sidebar-32.png",
+        },
+        @"open_at_install": @NO,
+    },
+};
+
+static auto *neitherSidebarActionNorPanelManifest = @{
+    @"manifest_version": @3,
+    @"name": @"Sidebar Test (neither)",
+    @"description": @"Sidebar Test (neither)",
+    @"version": @"1",
+
+    @"permissions": @[],
+    @"background": @{
+        @"scripts": @[ @"background.js", ],
+        @"type": @"module",
+        @"persistent": @NO,
+    },
+};
+
+// This test fixture allows us to use sidebarConfig (which enables the sidebar feature flag) without manually constructing one on each run
+class WKWebExtensionAPISidebarTest : public testing::Test {
```

Comment:
This should be `WKWebExtensionAPISidebar` (drop the Test suffix.) that way it is consistent with the other extension test categories.